bindings/runner/src/common.rs

```diff
@@ -510,6 +516,61 @@ where
     })
 }
 
+/// Get a randomized list of peers connected to the conductor in the `ctx` for a given cell.
+///
+/// Requires:
+/// - The [HolochainAgentContext] to have a valid `cell_id`. Consider calling [install_app] in your setup before using this function.
+///
+/// Call this function as follows:
+/// ```rust
+/// use holochain_types::prelude::ActionHash;
+/// use holochain_wind_tunnel_runner::prelude::{call_zome, HolochainAgentContext, HolochainRunnerContext, AgentContext, HookResult};
+///
+/// fn agent_behaviour(ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext>) -> HookResult {
+///     let peer_list = get_peer_list_randomized(ctx)?;
+///     println!("Connected peers: {:?}", peer_list);
+///     Ok(())
+/// }
+/// ```
+///
+/// Method:
+/// - calls `app_agent_info` on the websocket.
+/// - filters out self
+/// - shuffles the list
+pub fn get_peer_list_randomized<SV>(
+    ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<SV>>,
+) -> WindTunnelResult<Vec<AgentPubKey>>
+where
+    SV: UserValuesConstraint,
+{
+    let cell_id = ctx.get().cell_id();
+    let reporter: Arc<Reporter> = ctx.runner_context().reporter();
+    let admin_ws_url = ctx.runner_context().get_connection_string().to_string();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            let admin_client = AdminWebsocket::connect(admin_ws_url, reporter).await?;
+            // No more agents available to signal, get a new list.
+            // This is also the initial condition.
+            let agent_infos_encoded = admin_client
+                .agent_info(Some(cell_id.clone()))
```

Comment:
Really I think this should use the new app_agent_info interface, but this is good enough for now.